beacon_node/beacon_chain/src/light_client_server_cache.rs

```diff
@@ -140,6 +151,40 @@ impl<T: BeaconChainTypes> LightClientServerCache<T> {
                     signature_slot,
                     chain_spec,
                 )?);
+
+                let new_light_client_update = LightClientUpdate::new(
```

Comment:
I think you should not condition creating new `LightClientUpdate` on having a non-zero finalized_block_root. If the chain does not finalize for a full sync committee period we still need to produce them. Your `is_better_light_client_update` implementation covers the case of finalized vs non-finalized updates and will prioritize the finalized ones.

`LightClientUpdate` should be produced anytime we may be able to produce a better update

Reply:
got it, we should now be persisting non finalized updates